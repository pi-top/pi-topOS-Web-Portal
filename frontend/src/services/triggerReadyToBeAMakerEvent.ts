import axios from "axios";

import apiBaseUrl from "./apiBaseUrl";

export default async function triggerReadyToBeAMakerEvent() {
  await axios.post(
    `${apiBaseUrl}/onboarding-miniscreen-ready-to-be-a-maker`,
    {}
  );
}
