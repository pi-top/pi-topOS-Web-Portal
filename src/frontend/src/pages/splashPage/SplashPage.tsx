import React from "react";

import introScreen from "../../assets/images/intro-screen.png";
import styles from "./SplashPage.module.css";
import Layout from "../../components/layout/Layout";

import leaveMiniscreenAppBreadcrumb from "../../services/leaveMiniscreenAppBreadcrumb";

export type Props = {
  goToNextPage: () => void;
};

export default ({ goToNextPage }: Props) => {
  return (
    <Layout
      banner={{
        src: introScreen,
        alt: "intro-screen"
      }}
      prompt={
        <>
          Are you ready to be a <span className="green">maker</span>?
        </>
      }
      nextButton={{
        onClick: () => {
          leaveMiniscreenAppBreadcrumb()
            .catch(() => null)
            .then(goToNextPage)
        },
        label: 'Yes'
      }}
      className={styles.root}
    />
  );
};
