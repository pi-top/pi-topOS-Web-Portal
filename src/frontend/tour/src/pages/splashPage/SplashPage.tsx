import React from "react";

import summaryScreen from "../../assets/images/summary-screen.png";
import styles from "./SplashPage.module.css";
import Layout from "../../components/layout/Layout";

export type Props = {
  goToNextPage: () => void;
};

export default ({ goToNextPage }: Props) => {
  return (
    <Layout
      banner={{
        src: summaryScreen,
        alt: "intro-screen"
      }}
      prompt={
        <>
          All done, now let's start <span className="green">making</span>!
        </>
      }
      nextButton={{
        onClick: goToNextPage,
        label: "Let's Go"
      }}
      className={styles.root}
    />
  );
};
