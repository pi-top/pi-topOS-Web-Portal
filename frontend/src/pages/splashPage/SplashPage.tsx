import React, { useState } from "react";

import CheckBox from "../../components/atoms/checkBox/CheckBox";

import introScreen from "../../assets/images/intro-screen.png";
import styles from "./SplashPage.module.css";
import Layout from "../../components/layout/Layout";

import triggerReadyToBeAMakerEvent from "../../services/triggerReadyToBeAMakerEvent";
import { UserType } from "../../types/UserType";

export type Props = {
  goToNextPage: (userType: UserType) => void;
};

export default ({ goToNextPage }: Props) => {
  const [isSchoolUser, setIsSchoolUser] = useState(false);

  return (
    <Layout
      banner={{
        src: introScreen,
        alt: "intro-screen",
      }}
      prompt={
        <>
          Are you ready to <span className="green">start</span>?
        </>
      }
      nextButton={{
        onClick: () => {
          triggerReadyToBeAMakerEvent()
            .catch(() => null)
            .then(() =>
              goToNextPage(isSchoolUser ? UserType.School : UserType.Home)
            );
        },
        label: "Yes",
      }}
      className={styles.root}
    >
      <CheckBox
        name="user"
        label="I'm a school"
        checked={isSchoolUser}
        onChange={() => setIsSchoolUser(!isSchoolUser)}
        className={styles.checkbox}
      />
    </Layout>
  );
};
