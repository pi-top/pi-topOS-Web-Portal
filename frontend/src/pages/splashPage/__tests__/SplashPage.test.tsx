import React from "react";
import {
  render,
  fireEvent,
  BoundFunction,
  GetByBoundAttribute,
  QueryByText,
  GetByText,
  wait,
} from "@testing-library/react";


import triggerReadyToBeAMakerEvent from "../../../services/triggerReadyToBeAMakerEvent";
jest.mock("../../../services/triggerReadyToBeAMakerEvent");
const triggerReadyToBeAMakerEventMock = triggerReadyToBeAMakerEvent as jest.Mock;

import SplashPage, { Props } from "../SplashPage";

describe("SplashPage", () => {
  let defaultProps: Props;
  let languagePage: HTMLElement;
  let getByAltText: BoundFunction<GetByBoundAttribute>;
  let queryByText: BoundFunction<QueryByText>;
  let getByText: BoundFunction<GetByText>;
  beforeEach(() => {
    triggerReadyToBeAMakerEventMock.mockResolvedValue("OK");

    defaultProps = {
      goToNextPage: jest.fn(),
    };

    ({
      container: languagePage,
      getByAltText,
      queryByText,
      getByText,
    } = render(<SplashPage {...defaultProps} />));
  });

  it("renders correct image", () => {
    expect(getByAltText("intro-screen")).toMatchSnapshot();
  });

  it("renders prompt correctly", () => {
    const prompt = languagePage.querySelector(".prompt");
    expect(prompt).toMatchSnapshot();
  });

  it("renders Yes button", () => {
    expect(queryByText("Yes")).toBeInTheDocument();
  });

  it("calls triggerReadyToBeAMakerEvent on yes button click", () => {
    fireEvent.click(getByText("Yes"));

    expect(triggerReadyToBeAMakerEventMock).toHaveBeenCalled();
  });

  it("calls goToNextPage on yes button click", async () => {
    await wait();
    fireEvent.click(getByText("Yes"));

    await wait();
    expect(defaultProps.goToNextPage).toHaveBeenCalled();
  });

  describe('if triggerReadyToBeAMakerEvent service fails', () => {
    beforeEach(() => {
      triggerReadyToBeAMakerEventMock.mockRejectedValue(new Error("Oh oh, something happened"))
    });

    it("still calls goToNextPage", async () => {
      await wait();
      fireEvent.click(getByText("Yes"));

      await wait();
      expect(defaultProps.goToNextPage).toHaveBeenCalled();
    });
  })
});
