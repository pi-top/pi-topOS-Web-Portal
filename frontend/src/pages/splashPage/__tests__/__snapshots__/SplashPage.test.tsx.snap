// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplashPage renders correct image 1`] = `
<img
  alt="intro-screen"
  class="image"
  draggable="false"
  src="intro-screen.png"
  style="max-height: 100%; max-width: 100%;"
/>
`;

exports[`SplashPage renders prompt correctly 1`] = `
<h1
  class="prompt"
>
  Are you ready to be a 
  <span
    class="green"
  >
    maker
  </span>
  ?
</h1>
`;
